### PR TITLE
Created list contacts UI

### LINKED ISSUE #5: Create list contacts UI

### PR DESCRIPTION
**Pull request by anandhakrishnanaji**
Fixes #5

### CODE REVIEW

File: app/javascript/src/components/Common/Menu.jsx [hunk 1]
```diff
@@ -4,13 +4,11 @@ import { Search, Plus, Settings } from "neetoicons";
 import { Typography } from "neetoui";
 import { MenuBar } from "neetoui/layouts";
 
-import { MENU_BAR_BLOCKS } from "./constants";
-
-const Menu = ({ showMenu }) => {
+const Menu = ({ showMenu, menuBarBlocks }) => {
```
Comment (yedhink) [inline app/javascript/src/components/Common/Menu.jsx:7]: Why was this moved to the `Common` namespace? Are we using this component at multiple places within the app?

File: app/javascript/src/components/Dashboard/Contacts/Table.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+);
+
+const renderOptionButton = () => (
+  <Dropdown buttonStyle="text" icon={MenuHorizontal}>
+    <div className="p-1.5">
+      <li>Edit</li>
+      <li>Delete</li>
+    </div>
+  </Dropdown>
+);
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/Table.jsx:29]: @anandhakrishnanaji 
This is good abstraction. Can you please create a video and put it up in neeto-ui-challenge channel? That will help others. The video should cover these points:
- When do we need to make the judgement to move the JSX rendering logic into a separate function?
- What is the difference between writing this as a function vs writing it as a component? When to prefer which? We need to name it as a component when we need to make it a standalone component.
- Naming conventions for `render*` functions.
- We should also mention that we should pass in the arguments required for the working of the `render*` functions rather than taking them from the scope. The reason is that, then only we can easily move out these `render` functions into some other file when the current component becomes too big. You will have to first make related changes in your code before talking about this point.

File: app/javascript/src/components/Dashboard/Contacts/index.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+const Contacts = () => {
+  const [showMenuBar, setShowMenuBar] = useState(false);
+  const [searchTerm, setSearchTerm] = useState("");
+
+  return (
+    <>
+      <Menu menuBarBlocks={MENU_BAR_BLOCKS} showMenu={showMenuBar} />
+      <Container>
+        <Header
+          menuBarToggle={() => setShowMenuBar(!showMenuBar)}
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Contacts/index.jsx:22]: Go through the slack thread.

**Review by yedhink — COMMENTED** [review]
@anandhakrishnanaji

File: app/javascript/src/components/Common/Menu.jsx (hunk 1 above)
Comment (anandhakrishnanaji) [reply app/javascript/src/components/Common/Menu.jsx:7]: Yes, I have used the Menu component in both Notes and Contacts page.

File: app/javascript/src/components/Dashboard/Contacts/Table.jsx (hunk 2 above)
Comment (anandhakrishnanaji) [reply app/javascript/src/components/Dashboard/Contacts/Table.jsx:29]: Sure 👍

Comment (yedhink) [reply app/javascript/src/components/Dashboard/Contacts/Table.jsx:29]: PS: You should make the following changes:
- Rename `renderOptionButton` to `renderActionDropdown`. You are not simply rendering a single Button.
- Make the whole arguments of `renderAvatarWithNameAndRole` named arguments. Right now you have made the first argument a positional argument and the rest as named arguments, which is not right.

Comment (anandhakrishnanaji) [reply app/javascript/src/components/Dashboard/Contacts/Table.jsx:29]:
```
const renderAvatarWithNameAndRole = (_, {firstName, lastName, role }) => (
```
You meant like this right?
I am using the function with the Table component and it passes the key used as the first positional argument.

Comment (yedhink) [reply app/javascript/src/components/Dashboard/Contacts/Table.jsx:29]: @anandhakrishnanaji Ok. In that case we can stick onto the current argument format itself.
